### PR TITLE
Oauth2 extensions pili

### PR DESCRIPTION
**Pull request by mpguerra**

### CODE REVIEW

File: apicast/src/oauth/apicast_oauth.lua [hunk 1]
```diff
@@ -34,25 +34,23 @@ end
 
 function _M.extract_params()
   local params = {}
-  local header_params = ngx.req.get_headers()
+  local auth
 
-  params.authorization = {}
-
-  if header_params['Authorization'] then
-    params.authorization = re.split(ngx.decode_base64(re.split(header_params['Authorization']," ", 'oj')[2]),":", 'oj')
```
Comment (kevprice83) [inline apicast/src/oauth/apicast_oauth.lua]: Why is the base64 decode function removed here? http_authorization module does this?

File: apicast/src/oauth/apicast_oauth.lua [hunk 2]
```diff
@@ -78,13 +76,13 @@ function _M.respond_and_exit(status, body, headers)
   ngx.exit(ngx.HTTP_OK)
 end
 
-function _M.respond_with_error(status, message)
+function _M.respond_with_error(status, message, description)
```
Comment (kevprice83) [inline apicast/src/oauth/apicast_oauth.lua]: difference between message and description?

File: apicast/src/oauth/apicast_oauth.lua [hunk 3]
```diff
@@ -412,27 +392,26 @@ function _M.callback()
...
   end
-  
+
   code = ok
   ngx.header.content_type = "application/x-www-form-urlencoded"
   return ngx.redirect( client_data.redirect_uri .. "?code="..code.."&state=" .. (client_data.state or ""))
-
 end
 
 return _M
```
Comment (kevprice83) [inline apicast/src/oauth/apicast_oauth.lua]: newline necessary at the end of each file?

File: apicast/src/oauth/apicast_oauth.lua (hunk 1 above)
Comment (mpguerra) [inline apicast/src/oauth/apicast_oauth.lua]: correct

File: apicast/src/oauth/apicast_oauth.lua (hunk 2 above)
Comment (mpguerra) [inline apicast/src/oauth/apicast_oauth.lua]: see rfc: https://tools.ietf.org/html/rfc6749#section-5.2 message should really be called error as per the rfc and then you can also have error_description but this latter one is optional

File: apicast/src/oauth/apicast_oauth.lua (hunk 3 above)
Comment (mpguerra) [inline apicast/src/oauth/apicast_oauth.lua]: dunno, luacheck didn't complain :)

File: apicast/conf/nginx.conf [hunk 4]
```diff
@@ -4,6 +4,7 @@ env REDIS_URL;
 env RESOLVER;
 env BACKEND_ENDPOINT_OVERRIDE;
 env RHSSO_ENDPOINT;
+env APICAST_OAUTH_ACCESS_TOKEN_TTL;
```
Comment (mikz) [inline apicast/conf/nginx.conf]: All variables starting with `APICAST_` all forwarded  automatically.

**Pull request merged**